src/OpenFeature.Contrib.Providers.Flagsmith/FlagsmithProvider.cs

```diff
@@ -0,0 +1,185 @@
+﻿using Flagsmith;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Threading.Tasks;
+using Trait = Flagsmith.Trait;
+using OpenFeature.Error;
+using System.Globalization;
+
+namespace OpenFeature.Contrib.Providers.Flagsmith
+{
+    /// <summary>
+    /// FlagsmithProvider is the .NET provider implementation for the feature flag solution Flagsmith.
+    /// </summary>
+    public class FlagsmithProvider : FeatureProvider
+    {   
+        private readonly static Metadata Metadata = new("Flagsmith Provider");
+        delegate bool TryParseDelegate<T>(string value, out T x);
+        internal readonly IFlagsmithClient _flagsmithClient;
+
+        /// <summary>
+        /// Key that will be used as identity for Flagsmith requests. Default: "targetingKey"
+        /// </summary>
+        public string TargetingKey { get; set; } = "targetingKey";
+
+        /// <summary>
+        /// Determines whether to resolve a feature value as a boolean or use
+        /// the isFeatureEnabled as the flag itself. These values will be false
+        /// and true respectively.
+        /// Default: false
+        /// </summary>
+        public bool UsingBooleanConfigValue { get; set; }
```

Comment:
I think I'd prefer to see this in it's own config class? Perhaps we can create a `FlagsmithProviderConfiguration` class which extends the `FlagsmithConfiguration` class from the Flagsmith lib?

Reply:
done, but i created a separate one. Just to avoid additional dependency